README.md

```diff
@@ -415,45 +403,111 @@ This field SHOULD NOT be used to sign arbitrary data, such as signature challeng
 
-#### 3.2.7.2 Examples
+```json
+"https://example.com",
+"https://example.com/blog"
+```
+
+#### 3.2.5.3 Abilities
+
+Abilities MUST be presented as a string. By convention, abilities SHOULD be namespaced with a slash, such as `msg/send`. One or more abilities MUST be given for each resource.
+
+#### 3.2.5.4 Caveat Array
+
+Caveats MAY be open ended. Caveats MUST be understood by the executor of the eventual [invocation]. Caveats MUST prevent invocation otherwise.
+
+On validation, the caveat array MUST be treated as a logically disjunct (an "OR", NOT an "and"). In other words: passing validation against _any_ caveat in the array MUST be terated as valid. For example, consider the following capabilities:
+
+```json
+{
+  "dns:example.com?TYPE=TXT": {
+    "crud/create": "*"
+  },
+  "https://example.com/blog": {
+    "crud/read": "*",
+    "crud/update": [
+      {"status": "draft"},
+      {"status": "published", "day-of-week": "Monday"} // only publish on Mondays
+    ]
+  }
+}
+```
+
+The above MUST be interpreted as the set of capabilities below. If _any_ are matched, the check MUST pass validation.
+
+| Resource                   | Ability       | Caveat                                            |
+|----------------------------|---------------|---------------------------------------------------|
+| `dns:example.com?TYPE=TXT` | `crud/create` | Always                                            |
+| `https://example.com/blog` | `crud/read`   | Always                                            |
+| `https://example.com/blog` | `crud/update` | `{status: "draft"}`                               |
+| `https://example.com/blog` | `crud/update` | `{status: "publishsed", "day-of-week": "monday"}` |
+
+The caveat array SHOULD NOT be empty, as an empty array means "in no case" (which is equivalent to not listing the ability). This follows from the rule that delegations MUST be of equal or lesser scope. When an array is given, an attenuated caveat MUST (syntactically) include all of the fields of the relevant proof caveat, plus the newly introduced caveats.
+
+| Proof Caveats | Delegated Caveats | Is Valid? | Comment                              |
+|---------------|-------------------|-----------|--------------------------------------|
+| `"*"`         | `"*"`             | Yes       | Equal                                |
+| `[x]`         | `[x]`             | Yes       | Equal                                |
+| `[x]`         | `"*"`             | No        | Escalation to any                    |
+| `"*"`         | `[x]`             | Yes       | Introduces a new caveat              |
+| `[x, y]`      | `[x]`             | Yes       | Removes a capability                 |
+| `[x, y]`      | `[x, (y + z)]`    | Yes       | Attenuates existing caveat           |
+| `[x, y]`      | `[x, y, z]`       | No        | Escalation by adding new capability  |
```

Comment:
I ran into a problem. It was actually pretty counterintuitive to have `[]` mean "no caveats". Since we want to be able to express more than one _disjoint_ capabilty on the same resource/ability pair, this means that we have something less like a caveat and more like a capability or "opening" (i.e. matching "constructively").

Concretely, in UCAN 0.9 you can say this:

```json
[
  {
    "with": "https://example.com",
    "can": "crud/create",
    "nb": { "day": "monday", "status": "published" }
  },
  {
    "with": "https://example.com",
    "can": "crud/create",
    "nb": { "status": "draft" }
  }
]
```

i.e. you can create drafts at any time, but only publish on Mondays. In the new syntax:

```json
{
  "https://example.com": {
    "crud/create": [
      { "day": "monday", "status": "published" },
      {"status": "draft"}
    ]
  }
}
```

Which is itself straightforward. Now what does this mean?

```
```json
{
  "https://example.com": {
    "crud/create": []
  }
}
```

On the face of it, it means "no caveats". But this is strange! I can attenuate from `[a, b]` to `[a]`, which is a narrowing of permission scope. If I go from `[a]` to `[]`, it now _expands_ the scope! That's super counterintuitive! Here's the _old_ table:

| Proof Caveats | Delegated Caveats | Is Valid? | Comment |
|---------------|-------------------|-----------|--------|
| `[]`          | `[]`              | Yes       | Equal |
| `[a]`         | `[a]`             | Yes       | Equal |
| `[a]`         | `[]`              | No        | Escalation from some caveats to none |
| `[]`          | `[a]`             | Yes       | Introduces a new caveat |
| `[a, b]`      | `[a]`             | Yes       | Removes a capability     |
| `[a]`         | `[a, b]`          | Maybe     | Adds an attenuated capability |

The proposal in the current draft is to use `"*"` to mean "no caveats", and `[]` would mean "under no circumstances is this valid". If you look at the _new_ table (directly above this comment), it's much more consistent.

It's also more consistent with the `"*"` wildcard that we have in other places, such as `ucan:*`.

@Gozala and @dholms any thoughts?

Reply:
I also looked at `null` instead of `"*"`, but it was a bit confusing to read `"crud/create": null` as "unrestricted"

Reply:
How about we use `[{}]` to imply "no caveats" and treat `[]` as either:

1. Absence of delegation (which is I think what you're making a case for). In other words treat it same as `null` or as if key was not even there.
2. Alternatively it could mean inherit whatever caveats are in the proofs. I suspect this would be more error prone and probably not a good idea, but thought it was worth mentioning anyway.

I don't really like `"*"` because it introduces special string where maps are expected, `[{}]` on the other hand is basically what one would write if there were no caveats.

Reply:
Okay yeah `[{}]`is logical 👍

Reply:
Done!